proto/sequencerblockapis/astria/sequencerblock/v1/block.proto

```diff
@@ -151,6 +151,24 @@ message RollupData {
   }
 }
 
+message UpgradeChangeHashesWithProof {
+  // The SHA256 digests of all upgrade changes applied during this block, if an upgrade was
+  // activated at this height.
+  repeated bytes upgrade_change_hashes = 1;
```

Comment:
why have multiple hashes instead of just one?

Reply:
It makes it more fine-grained, which potentially makes it simpler for external consumers to discover whether a particular upgrade change has been applied or not.  It might never be used, but I think it would be a bit painful to retrofit this functionality, whereas it's pretty trivial to add it now.